### PR TITLE
Add decrypt function

### PR DESCRIPTION
**Pull request by jakekgrog**
# Why
- Add `decrypt()` function
- Add E2E tests for `decrypt()` function

# How
- Add them

### CODE REVIEW

File: evervault-core/src/commonMain/kotlin/com/evervault/sdk/core/models/FunctionRunResponse.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.evervault.sdk.core.models
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class FunctionResponseResult(
```
Comment (jeremywagemans) [inline evervault-core/src/commonMain/kotlin/com/evervault/sdk/core/models/FunctionRunResponse.kt:6]: I don't think this is accurate. Why do we want to limit the output of a function to only return a message?

File: evervault-core/src/commonMain/kotlin/com/evervault/sdk/Evervault.kt [hunk 2]
```diff
@@ -95,6 +95,66 @@ class Evervault private constructor() {
...
+     * ## Example
+     * ```kotlin
+     * val decrypted = Evervault.shared.decrypt<String>("token1234567890", encryptedData)
+     * ```
+     *
+     * The `decrypt()` function allows you to decrypt previously encrypted data using a token and attempt to deserialize it to the parameterized type. The token is a single use, time bound token for decrypting data.
+     *
+     * Tokens will only last for 5 minutes and must be used with the same payload that was used to create the token.
+     *
+     * The function returns the decrypted data as `Any`, and the caller is responsible for safely casting the result based on the original data type. For Boolean, Numerics, and Strings, the encrypted data is returned as a String. For Lists and Maps, the encrypted data maintains the same structure but is encrypted. For ByteArray, the encrypted data is returned as encrypted ByteArray.
```
Comment (nassir90) [inline evervault-core/src/commonMain/kotlin/com/evervault/sdk/Evervault.kt:152]: Should this say "the encrypted data is returned as a decrypted ByteArray"?

File: evervault-core/src/commonMain/kotlin/com/evervault/sdk/core/HttpRequest.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        val responseBody = response.bodyAsText()
+        val body = json.decodeFromString<FunctionRunResponse>(responseBody)
+        return body
+    }
+
+    private suspend fun <T> executeDecryptWithToken(token: String, data: Any): T {
+        val response: HttpResponse = httpClient.post("${config.apiUrl}/decrypt") {
+            setBody(data)
+            headers {
+                append("Authorization", "Bearer ${token}")
```
Comment (nassir90) [inline evervault-core/src/commonMain/kotlin/com/evervault/sdk/core/HttpRequest.kt:98]:
```suggestion
                append("Authorization", "Token ${token}")
```

Pretty sure that this should use the `Token` scheme due to this: https://github.com/evervault/oxide/blob/a5a0448476323e2809e078a99ae8c48581d631f3/crates/ev-api-gateway/src/services/auth/token_auth.rs#L68.

Comment (e-moran) [reply evervault-core/src/commonMain/kotlin/com/evervault/sdk/core/HttpRequest.kt:98]: Yes! This should be `Token` not `Bearer`

Comment (jakekgrog) [reply evervault-core/src/commonMain/kotlin/com/evervault/sdk/core/HttpRequest.kt]: Good catch!

**Pull request merged**